### PR TITLE
[PHR-125] Added record-type counts

### PR DESCRIPTION
**Pull request by jdkizer9**

### CODE REVIEW

File: app/src/main/java/org/thecommonsproject/android/commonhealth/sampleapp/MainViewModel.kt [hunk 1]
```diff
@@ -84,6 +121,41 @@ class MainViewModel(
...
+                    )
+                )
+            }
+        }
+        jobs.joinAll()
+
+
+        val endTime = Date()
+        val totalDuration = (endTime.time - startTime.time).toDouble() / 1000.0
+        val avgDuration = totalDuration / typesToFetch.count()
```
Comment (adriansoghoian) [inline app/src/main/java/org/thecommonsproject/android/commonhealth/sampleapp/MainViewModel.kt]: Do we want to leave the profiling code in this sample app, or the one in Android-PHR repo? Do we think it's helpful for others, too?

File: app/src/main/java/org/thecommonsproject/android/commonhealth/sampleapp/fragments/CategoryListFragment.kt [hunk 2]
```diff
@@ -171,14 +178,21 @@ class CategoryListFragment : Fragment() {
...
     }
 
     class ContentAdapter(
-        private val categories: List<DataType>,
+        private val categories: List<DataType.ClinicalResource>,
         private val generateOnClickListener: (DataType) -> View.OnClickListener
     ) : RecyclerView.Adapter<CategoryListItemViewHolder>() {
 
+        var resultCounts: Map<DataType.ClinicalResource, Int>? = null
+        fun updateResoutCounts(newResultCounts: Map<DataType.ClinicalResource, Int>) {
```
Comment (adriansoghoian) [inline app/src/main/java/org/thecommonsproject/android/commonhealth/sampleapp/fragments/CategoryListFragment.kt]: Is the function name supposed to be `resultsCounts`?

File: app/src/main/java/org/thecommonsproject/android/commonhealth/sampleapp/fragments/ResourceListFragment.kt [hunk 3]
```diff
@@ -86,22 +84,18 @@ class ResourceListFragment : Fragment() {
         recyclerView.layoutManager = LinearLayoutManager(requireContext())
 
         val progressBar = view.findViewById<ProgressBar>(R.id.progress_bar)
-        progressBar.visibility = View.VISIBLE
-        viewModel.viewModelScope.launch {
-            if (!viewModel.isCommonHealthAvailable(requireContext())) {
-                Toast.makeText(requireContext(), "Please make sure CommonHealth is installed and setup", Toast.LENGTH_LONG).show()
-                return@launch
+        progressBar.visibility = View.GONE
+//        viewModel.viewModelScope.launch {
```
Comment (adriansoghoian) [inline app/src/main/java/org/thecommonsproject/android/commonhealth/sampleapp/fragments/ResourceListFragment.kt]: I'd probably just remove this entirely if we don't want it anymore?

Comment (jdkizer9) [reply app/src/main/java/org/thecommonsproject/android/commonhealth/sampleapp/fragments/ResourceListFragment.kt]: done

File: app/src/main/java/org/thecommonsproject/android/commonhealth/sampleapp/fragments/CategoryListFragment.kt (hunk 2 above)
Comment (jdkizer9) [reply app/src/main/java/org/thecommonsproject/android/commonhealth/sampleapp/fragments/CategoryListFragment.kt]: Renamed to `resultsCounts`

File: app/src/main/java/org/thecommonsproject/android/commonhealth/sampleapp/MainViewModel.kt (hunk 1 above)
Comment (jdkizer9) [reply app/src/main/java/org/thecommonsproject/android/commonhealth/sampleapp/MainViewModel.kt]: Removed this

**Comment by jdkizer9** [comment]
@adriansoghoian this is ready to review again

**Review by adriansoghoian — APPROVED** [review]

**Pull request merged**